src/hotspot/share/opto/superword.hpp

```diff
@@ -424,7 +426,94 @@ class SuperWord : public ResourceObj {
   // Combine packs A and B with A.last == B.first into A.first..,A.last,B.second,..B.last
   void combine_pairs_to_longer_packs();
 
-  void split_packs_longer_than_max_vector_size();
+  // The lambda method for split_packs can return one of these tasks:
+  // 1) reject   -> the pack is removed
+  // 2) no_split -> split_size = 0, the old_pack is unchanged
+  // 3) else     -> split away split_size nodes from the end of the old_pack
+  class SplitTask {
+  private:
+    const uint _split_size;
+    const bool _reject;
```

Comment:
Should this be enum with 3 states you listed in comment?

Reply:
Refactored it :)